torch/nested/_internal/ops.py

```diff
@@ -46,16 +47,30 @@ def _wrap_jagged_dims(ndim, dims, op_name):
     # map to inner dims so we can print a nicer error message.
     zero_in_dims = 0 in wrapped_dims
     one_in_dims = 1 in wrapped_dims
-    if zero_in_dims ^ one_in_dims:
-        apply, not_apply = ("batch", "ragged") if zero_in_dims else ("ragged", "batch")
-        raise RuntimeError(
-            f"{op_name}(): applying over the {apply} dimension, but not the {not_apply}"
-            " dimension is not supported for NestedTensor"
+    if op_name in SUPPORTED_JAGGED_REDUCTIONS:
```

Comment:
nit: instead of maintaining a separate list of `SUPPORTED_JAGGED_REDUCTIONS`, I think I'd prefer one or two kwargs conceptually customizing `_wrap_jagged_dim()`'s behavior. I think the cases we want to support are:
1. pre-PR case: allow everything except `batch but not ragged` or `ragged but not batch` dims
2. new case: allow everything except batch-only

I think we can come up with a nice set of kwargs that will cover those cases :)

Another suggestion is to perhaps have `_wrap_jagged_dim()` return `(wrapped dim, operating_on_batch, operating_on_ragged)` to provide flexibility to the caller on how to handle these. as-is, the return values aren't super intuitive

Reply:
agreed, and i think this follows a point made in a future comment as well! i've made some modifications to the return values from `_wrap_jagged_dims` to make the logic clearer and have also cleaned up the implementation of `sum` for `NestedTensor`